File: app/src/main/java/com/adsamcik/signalcollector/utility/FirebaseAssist.java
package com.adsamcik.signalcollector.utility;


import android.content.Context;
import android.support.annotation.NonNull;

import com.google.firebase.analytics.FirebaseAnalytics;

public class FirebaseAssist {
	public static final String STOP_TILL_RECHARGE_EVENT = "stop_till_recharge";
	public static final String STOP_EVENT = "notification_stop";
	public static final String MANUAL_UPLOAD_EVENT = "manual_upload";

	public static final String PARAM_SOURCE = "source";

	public static final String autoTrackingString = "auto_tracking";
	public static final String autoUploadString = "auto_upload";
	public static final String uploadNotificationString = "upload_notifications";

	public static void updateValue(@NonNull Context context, @NonNull String name, @NonNull String value) {
		FirebaseAnalytics.getInstance(context).setUserProperty(name, value);
	}
}

File: app/src/main/java/com/adsamcik/signalcollector/utility/FirebaseAssist.kt
package com.adsamcik.signalcollector.utility


import android.content.Context

import com.google.firebase.analytics.FirebaseAnalytics

object FirebaseAssist {
    const val STOP_TILL_RECHARGE_EVENT = "stop_till_recharge"
    const val STOP_EVENT = "notification_stop"
    const val MANUAL_UPLOAD_EVENT = "manual_upload"
    const val CLEARED_DATA_EVENT = "cleared_data"

    const val PARAM_SOURCE = "source"

    const val autoTrackingString = "auto_tracking"
    const val autoUploadString = "auto_upload"
    const val uploadNotificationString = "upload_notifications"

    fun updateValue(context: Context, name: String, value: String) {
        FirebaseAnalytics.getInstance(context).setUserProperty(name, value)
    }
}
